python/tk_framework_desktopserver/shotgun/scripts/cache_commands.py

```diff
@@ -257,7 +255,25 @@ def cache(
     with open(arg_data_file, "rb") as fh:
         arg_data = cPickle.load(fh)
 
-    sys.path = sys.path + arg_data["sys_path"]
+    # We do this in two steps. The first is to prepend to sys.path
+    # before immediately importing sgtk. After that, we replace the
+    # prepend with an append. This allows us to guarantee that we
+    # have sgtk imported from the right place, followed by setting
+    # up sys.path the way that is safest for other imports.
+    original_sys_path = copy.copy(sys.path)
+
+    try:
+        # The RPC api that spawned us has ensured that the first path
+        # in the sys.path list that it provided us is the root path that
+        # will get us the tk-core that we want. We can't prepend any more
+        # than that, because we might be running in a different version
+        # of Python in this process than SG Desktop is. We don't want to
+        # force Python 2.7 standard library modules to be imported if
+        # we're in Python 2.6 here, as an example.
+        sys.path = [arg_data["sys_path"][0]] + sys.path
```

Comment:
for clarity, would it make sense that `arg_data["sys_path"][0]` was sent down as a separate key, e.g. `arg_data["sgtk_path"]` or something like that?